File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # hash table
        n = len(nums)
        ans = 0
        occ = 0
        hash_table = {}
        for x in nums:
            hash_table[x] = hash_table.get(x,0)+1
        for x,y in hash_table.items():
            if y>= int(n/2):
                if(y>= occ):
                    occ = y
                    ans = x
        return ans

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // majority element is the element appearing more than [n/2]
        int ans = -1;
        int n = nums.size();
        map<int,int> mp;
        for(auto x:nums){
            mp[x]++;
            if(mp[x] > n/2) ans = x;
        }
        return ans;
    }
};